lib/entry_point.c

```diff
@@ -0,0 +1,80 @@
+#define LIBSBCL_LIBRARIAN_API_BUILD
+#ifdef __linux__
+#define _GNU_SOURCE
+#endif
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "sbcl_librarian.h"
+
+#ifdef _WIN32
+# include <Windows.h>
+# include <Process.h>
+# include <psapi.h>
+
+__thread intptr_t fatal_lisp_error_handler[5];
+#else
+# include <dlfcn.h>
+
+__thread jmp_buf fatal_lisp_error_handler;
+#endif
+
+#define BUF_SIZE 1024
+
+extern char *dir_name(char *path);
+extern void set_lossage_handler(void (*handler)(void));
+extern int initialize_lisp(int argc, const char *argv[], char *envp[]);
+
+int fatal_sbcl_error_occurred = 0;
+
+static void return_from_lisp(void)
+{
+    fatal_sbcl_error_occurred = 1;
+    fflush(stdout);
+    fflush(stderr);
+    longjmp(fatal_lisp_error_handler, 1);
+}
+
+static void do_initialize_lisp(const char *libsbcl_librarian_path)
+{
+    char *libsbcl_librarian_dir = dir_name(libsbcl_librarian_path);
+    int libsbcl_librarian_dir_len = strlen(libsbcl_librarian_dir);
+    int core_path_size = libsbcl_librarian_dir_len + sizeof("sbcl_librarian.core") + 1;
+    char *core_path = malloc(core_path_size);
+
+    snprintf(core_path, core_path_size, "%ssbcl_librarian.core", libsbcl_librarian_dir);
+
+    const char *init_args[] = {"", "--dynamic-space-size", "8192", "--core", core_path, "--noinform", "--no-userinit"};
+
+#ifdef __linux__
+    dlopen("libsbcl.so", RTLD_NOW | RTLD_GLOBAL);
+    dlopen(libsbcl_librarian_path, RTLD_GLOBAL | RTLD_NOW);
+#endif
```

Comment:
Add note on role in making symbols visible

Reply:
done